lib/Parse/ParseExpr.cpp

```diff
@@ -2420,8 +2420,8 @@ parseClosureSignatureIfPresent(SmallVectorImpl<CaptureListEntry> &captureList,
   // Extract names of the tuple elements and preserve the structure
   // of the tuple (with any nested tuples inside) to be able to use
   // it in the fix-it without any type information provided by user.
-  std::function<StringRef(const TypeRepr *)> getTupleNames =
-      [&](const TypeRepr *typeRepr) -> StringRef {
+  std::function<std::string (const TypeRepr *)> getTupleNames =
```

Comment:
It's a good temporary fix, but I think this code could use some deeper refactoring than just this.  The lifetime of the string really is just shy of local to this lambda.  I don't think all the extra effort of recursively opening then flushing stream buffers is worth it in the long run.  This should be pulled out into a `static` function that takes the fixit's stream as a parameter.